### PR TITLE
Remove old payment gateway task components

### LINKED ISSUE #7216: Clean up old payment gateway task
With the payment gateway suggestions feature in place, we no longer need the old payments task.

We should remove these files to lower maintenance costs on housing multiple features and prevent updates/fixes from being handled in the wrong components.

This issue will also require enabling the new payment gateway suggestions feature and making sure the task list degrades gracefully if this feature is not enabled.

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #7216 

* Removes old payment gateway task components
* Cleans up component folder organization
* Turns on the PaymentGatewaySuggestions feature in all environments

### Screenshots

Screen Shot 2021-06-22 at 7 44 03 PM

### Detailed test instructions:

1. Smoke test the payments task
2. Build the plugin zip and make sure the feature is enabled and working
3. Optionally, turn off the `payment-gateway-suggestions` feature flag and note that it does not appear in the task list

### CODE REVIEW

**Review by louwie17 — COMMENTED** [review]
There still appears to be some build errors with imports and usage of deleted files:
`getPaymentMethods` - client/header/activity-panel/panels/help.js
`isWCPaySupported` - ./client/payments/payment-recommendations.tsx
`WCPayUsageModal` - /client/layout/index.js
`isWCPaySupported` - ../selective-bundle/selective-extensions-bundle/index.js

**Comment by joshuatf** [comment]
Thanks for taking a look, @louwie17!  Fixed those references and updated the retrieval of gateway suggestions in https://github.com/woocommerce/woocommerce-admin/pull/7224/commits/a69e2ec0b406909e652704f196d2e780dedb8480.

**Comment by joshuatf** [comment]
@louwie17  This is ready for another look.  The failing test on this was the BACS setup task.  I believe this is due to the issue with SlotFill not working during tests.  Can you confirm @joelclimbsthings?

In non-headless mode, all tests pass, but in headless mode, that one test fails.  I hate to remove this test, but not quite sure how to get around this failure.

File: client/header/activity-panel/panels/help.js [hunk 1]
```diff
@@ -123,53 +101,53 @@ function getPaymentsItems( props ) {
...
 				'https://docs.woocommerce.com/document/klarna-payments/?utm_source=help_panel',
 		},
-		showPayFast && {
+		paymentGatewaySuggestions.payfast && {
 			title: __( 'PayFast Setup and Configuration', 'woocommerce-admin' ),
 			link:
 				'https://docs.woocommerce.com/document/payfast-payment-gateway/?utm_source=help_panel',
 		},
-		showEway && {
+		paymentGatewaySuggestions.eway_payments && {
```
Comment (louwie17) [inline client/header/activity-panel/panels/help.js:150]: I assume this should be changed to `eway` in relation to this PR: https://github.com/woocommerce/woocommerce-admin/pull/7228

**Review by louwie17 — COMMENTED** [review]
Nice work on this @joshuatf I am enjoying this very much:
Screen Shot 2021-06-24 at 5 29 38 PM
It will feel nice to have most of the payment stuff pushed out onto the plugins versus our repo :)
Everything tested well, there is just one small change for the eway help item, oooh and don't forget to add a changelog for this 😄

File: client/header/activity-panel/panels/help.js (hunk 1 above)
Comment (joshuatf) [reply client/header/activity-panel/panels/help.js]: Good catch!  Updated.

**Comment by joshuatf** [comment]
Thanks for taking another look, @louwie17.  Addressed the `eway` change and rebased and added a changelog entry.

**Comment by joelclimbsthings** [comment]
> This is ready for another look. The failing test on this was the BACS setup task. I believe this is due to the issue with SlotFill not working during tests. Can you confirm @joelclimbsthings?

Looks like tests are passing, so this is resolved?

File: readme.txt [hunk 2]
```diff
@@ -79,20 +79,21 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
...
-- Tweak: Revert Card component removal #7167
 - Fix: Currency display on Orders activity card on homescreen #7181
 - Fix: Report export filtering bug. #7165
 - Fix: Localize string for description #7219
 - Fix: Use tab char for the CSV injection prevention. #7154
 - Fix: Use saved form values if available when switching tabs #7226
 - Fix: The use of gridicons for Analytics section controls. #7237
-- Dev: Add `woocommerce_admin_export_id` filter for customizing the export file name #7178
+- Tweak: Revert Card component removal #7167
 - Update: Notes to use a date range. #7222
```
Comment (louwie17) [inline readme.txt:97]: Thanks for re-arranging these, it be nice to keep them in alphabetical order as it prevents conflicts 😄

**Review by louwie17 — APPROVED** [review]
Thanks for the update @joshuatf the Eway gateway tested well this time 👍 👍

**Pull request merged**